iceoryx_posh/test/moduletests/test_popo_event_variable.cpp

```diff
@@ -0,0 +1,247 @@
+{
+    for (auto& notification : m_eventVarData.m_activeNotifications)
+    {
+        EXPECT_THAT(notification, Eq(false));
+    }
+}
+
+TEST_F(EventVariable_test, NotifyActivatesCorrectIndex)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 1U;
+    EventNotifier sut(m_eventVarData, index);
+    sut.notify();
+    for (uint64_t i = 0U; i < iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET; i++)
+    {
+        if (i == index)
+        {
+            EXPECT_THAT(m_eventVarData.m_activeNotifications[i], Eq(true));
+        }
+        else
+        {
+            EXPECT_THAT(m_eventVarData.m_activeNotifications[i], Eq(false));
+        }
+    }
+}
+
+TEST_F(EventVariable_test, NotifyActivatesNoIndexIfIndexIsTooLarge)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET;
+    EventNotifier sut(m_eventVarData, index);
+    sut.notify();
+    for (const auto& notification : m_eventVarData.m_activeNotifications)
+    {
+        EXPECT_THAT(notification, Eq(false));
+    }
+}
+
+TEST_F(EventVariable_test, GetCorrectNotificationVectorAfterNotifyAndWait)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 1U;
+    EventNotifier notifier(m_eventVarData, index);
+    EventListener listener(m_eventVarData);
+
+    notifier.notify();
+    const auto& activeNotifications = listener.wait();
+    ASSERT_THAT(activeNotifications.size(), Eq(1U));
+    EXPECT_THAT(activeNotifications[0], Eq(index));
+}
+
+TEST_F(EventVariable_test, GetCorrectNotificationVectorAfterMultipleNotifyAndWait)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 1U;
+    EventNotifier notifier1(m_eventVarData, index);
+    EventNotifier notifier2(m_eventVarData, 0U);
+    EventListener listener(m_eventVarData);
+
+    notifier1.notify();
+    notifier2.notify();
+    const auto& activeNotifications = listener.wait();
+    ASSERT_THAT(activeNotifications.size(), Eq(2U));
+    EXPECT_THAT(activeNotifications[0], Eq(0U));
+    EXPECT_THAT(activeNotifications[1], Eq(index));
+}
+
+TEST_F(EventVariable_test, WaitAndNotifyResultsInCorrectNotificationVector)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 5U;
+    EventNotifier notifier(m_eventVarData, index);
+    EventListener listener(m_eventVarData);
+    notificationVector activeNotifications;
+
+    std::thread waiter([&] {
+        activeNotifications = listener.wait();
+        ASSERT_THAT(activeNotifications.size(), Eq(1U));
+        EXPECT_THAT(activeNotifications[0], Eq(index));
+    });
+
+    notifier.notify();
+    waiter.join();
+}
+
+TIMING_TEST_F(EventVariable_test, WaitBlocks, Repeat(5), [&] {
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 5U;
+    EventNotifier notifier(m_eventVarData, index);
+    EventListener listener(m_eventVarData);
+    notificationVector activeNotifications;
+    iox::posix::Semaphore semaphore =
```

Comment:
please give the semaphore a meaningful name, e.g. `threadSetupSemaphore`

Reply:
done